test/e2e/specs/editor/blocks/avatar.spec.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * WordPress dependencies
+ */
+
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Avatar', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllUsers();
+	} );
+
+	test.beforeEach( async ( { admin, requestUtils } ) => {
+		await requestUtils.createUser( {
+			username: 'user',
+			email: 'gravatartest@gmail.com',
+			firstName: 'Gravatar',
+			lastName: 'Gravatar',
+			roles: [ 'author' ],
+			password: 'gravatargravatar123magic',
+		} );
+
+		await admin.createNewPost();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllUsers();
+	} );
+
+	test( 'should change image when user is changed', async ( {
+		editor,
+		page,
+	} ) => {
+		// Inserting a quote block
+		await editor.insertBlock( {
+			name: 'core/avatar',
+		} );
+
+		const username = 'Gravatar Gravatar';
+
+		const avatarBlock = page.locator(
+			'role=document[name="Block: Avatar"]'
+		);
+
+		await expect( avatarBlock ).toBeVisible();
+
+		const avatarImage = avatarBlock.locator( 'img' );
+
+		await expect( avatarImage ).toBeVisible();
+
+		const originalSrc = await avatarImage.getAttribute( 'src' );
+
+		const blockInspectorControls = page.locator(
+			'.block-editor-block-inspector'
+		);
+
+		await expect( blockInspectorControls ).toBeVisible();
+
+		const userComboBox = blockInspectorControls.locator(
+			'.components-combobox-control'
+		);
+
+		await expect( userComboBox ).toBeVisible();
+
+		const userInput = userComboBox.locator(
+			'.components-combobox-control__input'
+		);
+
+		await expect( userInput ).toBeVisible();
```

Comment:
Generally, there's no need to worry about all the intermediate elements being visible. An action like `click` will implicitly check that an element is visible.

Another thing is that it's good to prefer role selectors over css selectors. This practice will go some way to helping make sure Gutenberg is using accessible HTML.

You should be able to achieve the same sort of thing by doing:
```js
const userInput = page.locator( 'role=region[name="Editor settings"i] >> role=combobox[name="User"i]' );
await userInput.click()
```

The accessibility inspector in the browser dev tools is pretty good for figuring this stuff out:
![Screen Shot 2022-08-10 at 2 19 00 pm](https://user-images.githubusercontent.com/677833/183829383-b19ec6f4-9bed-4fcf-8911-ea086f2e8821.png)

Reply:
Sounds good, thank you for the info.